mlrun/projects/project.py

```diff
@@ -1797,28 +1797,126 @@ def set_model_monitoring_application(
-        :param handler:                 default function handler to invoke (can only be set with .py/.ipynb files)
-        :param with_repo:               add (clone) the current repo to the build source
-        :param tag:                     function version tag (none for 'latest', can only be set with .py/.ipynb files)
+        :param handler:                 Default function handler to invoke (can only be set with .py/.ipynb files)
+        :param with_repo:               Add (clone) the current repo to the build source
+        :param tag:                     Function version tag (none for 'latest', can only be set with .py/.ipynb files)
                                         if tag is specified and name is empty, the function key (under the project)
                                         will be enriched with the tag value. (i.e. 'function-name:tag')
-        :param requirements:            a list of python packages
-        :param requirements_file:       path to a python requirements file
+        :param requirements:            A list of python packages
+        :param requirements_file:       Path to a python requirements file
         :param application_class:       Name or an Instance of a class that implementing the monitoring application.
         :param application_kwargs:      Additional keyword arguments to be passed to the
                                         monitoring application's constructor.
         """
 
+        function_object: RemoteRuntime = None
+        (
+            resolved_function_name,
+            function_object,
+            func,
+        ) = self._create_model_monitoring_function_helper(
+            func,
+            application_class,
+            name,
+            image,
+            handler,
+            with_repo,
+            tag,
+            requirements,
+            requirements_file,
+            **application_kwargs,
+        )
+        models_names = "all"
+        function_object.set_label(
+            mm_constants.ModelMonitoringAppLabel.KEY,
+            mm_constants.ModelMonitoringAppLabel.VAL,
+        )
+        function_object.set_label("models", models_names)
+
+        if not mlrun.mlconf.is_ce_mode():
+            function_object.apply(mlrun.mount_v3io())
+
+        # save to project spec
+        self.spec.set_function(resolved_function_name, function_object, func)
+
+        return function_object
+
+    def create_model_monitoring_function(
+        self,
+        func: str = None,
+        application_class: typing.Union[str, ModelMonitoringApplication] = None,
+        name: str = None,
+        image: str = None,
+        handler=None,
+        with_repo: bool = None,
+        tag: str = None,
+        requirements: typing.Union[str, typing.List[str]] = None,
+        requirements_file: str = "",
+        **application_kwargs,
+    ) -> mlrun.runtimes.BaseRuntime:
+        """
+        create a monitoring function object without setting it to the project
```

Comment:
Why is this needed? what's the use case? Add it to the description?
```suggestion
        Create a monitoring function object without setting it to the project
```